app/models/mdm/module/detail.rb

```diff
@@ -191,7 +191,69 @@ class Mdm::Module::Detail < ActiveRecord::Base
   #   {#supports_stance? support stances}.
   #
   #   @return ['active', 'passive', nil]
-
+  
+  #
+  # Scopes
+  #
+  
+  scope :module_author, ->(values) {
+    joins(Mdm::Module::Detail.join_association(:authors, Arel::Nodes::OuterJoin)).
+    where(
+      Mdm::Module::Author[:email].matches_any(values).or(
+        Mdm::Module::Author[:name].matches_any(values)
+      )
+    )
+  }
+  
+  scope :module_name, ->(values) {
+    where(
+      Mdm::Module::Detail[:fullname].matches_any(values).or(
+        Mdm::Module::Detail[:name].matches_any(values)
+      )
+    )
+  }
+  
+  scope :module_os_or_platform, ->(values) {
+    joins(
+      Mdm::Module::Detail.join_association(:platforms, Arel::Nodes::OuterJoin),
+      Mdm::Module::Detail.join_association(:targets, Arel::Nodes::OuterJoin)
+    ).where(
+      Mdm::Module::Platform[:name].matches_any(values).or(
+        Mdm::Module::Target[:name].matches_any(values)
+      )
+    )
+  }
+  
```

Comment:
Also, platform is fairly normalized (though it is a string, there should be a limited number of strings), while Target's text can be largely free-form. It doesn't seem good to merge normalized and unnormalized data into a single scope.